src/dmtcpplugin.cpp

```diff
@@ -130,6 +131,15 @@ EXTERNC int dmtcp_checkpoint()
   return rv;
 }
 
+EXTERNC int dmtcp_checkpoint_ckptDir(const char* ckptDir)
+{
+  if (ckptDir != NULL) {
+    CoordinatorAPI::instance().updateCkptDirGlobally(ckptDir);
+  }
+  int rv = dmtcp_checkpoint();
```

Comment:
Unless I'm misunderstanding the purpose of the new API, I think we don't need to call `dmtcp_checkpoint()` here. A user can make an explicit call to `dmtcp_checkpoint()` when they want.